### ISSUE TITLE
Slow Indigo Woodpecker - Borrowers cannot earn Uniswap V3 liquidity incentives on their positions

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-10-aloe-judging#59]
Slow Indigo Woodpecker

medium

# Borrowers cannot earn Uniswap V3 liquidity incentives on their positions
## Summary

Aloe Protocol only allows interacting directly with the UniswapV3 pools because all the opened positions must be owned by the Borrower contract.

Usually, when providing liquidity to UniswapV3 pool through the [`NonfungiblePositionManager`](https://github.com/Uniswap/v3-periphery/blob/main/contracts/NonfungiblePositionManager.sol) contract, the liquidity provider receives a NFT token which represents his liquidity position. 
This NFT token can then be staked in [`UniswapV3Staker`](https://docs.uniswap.org/contracts/v3/reference/periphery/staker/Design) contracts to earn liquidity mining incentives (in case a Pool offers such incentives).
Thus, the borrowers are missing out on all the liquidity mining incentives.

## Vulnerability Detail

When providing liquidity to UniswapV3 pool through the [NonfungiblePositionManager](https://github.com/Uniswap/v3-periphery/blob/main/contracts/NonfungiblePositionManager.sol), the liquidity provider receives a NFT token which represents his liquidity position. 
This NFT token can be staked in [UniswapV3Staker](https://docs.uniswap.org/contracts/v3/reference/periphery/staker/Design) contract to earn liquidity mining incentives.

Aloe protocol only allows interacting directly with the UniswapV3 pool because all the opened positions must be owned by the Borrower contract.
This can be observed through the [`_getAssets`](https://github.com/aloelabs/aloe-ii/blob/c71e7b0cfdec830b1f054486dfe9d58ce407c7a4/core/src/Borrower.sol#L502) function, which only takes into account the positions owned by the `Borrower` contract.

For example if there was USDC <-> BTC pool with a liquidity mining program for providing liquidity, the current Aloe protocol could not get these rewards.

## Impact

Borrowers in Aloe Protocol are missing out on the liquidity mining incentives. Since it's common for projects to incentive liquidity provision on their pools through [`UniswapV3Staker`](https://docs.uniswap.org/contracts/v3/guides/liquidity-mining/overview), this is a significant loss for the borrowers. 

## Code Snippet

https://github.com/aloelabs/aloe-ii/blob/c71e7b0cfdec830b1f054486dfe9d58ce407c7a4/core/src/Borrower.sol#L502

## Tool used

Manual Review

## Recommendation

Include all the NFTs minted through `NonfungiblePositionManager` contract, staked inside `UniswapV3Staker` contract,
and owned by the Borrower contract to the total assets accounting in the `_getAssets` function.
In other words it should be possible for the `Borrower` to earn liquidity incentives on his borrowed funds while still having the value of these incentivized Uniswap positions count toward his health level.
